lib/workers/repository/process/write.ts

```diff
@@ -40,7 +47,29 @@ export async function writeUpdates(
     }
     addMeta(meta);
     const branchExisted = branchExists(branch.branchName);
-    const res = await processBranch(branch);
+    let branchCache = {} as BranchCache;
+    if (branchExisted) {
+      branchCache =
+        cachedBranches?.find((br) => br.branchName === branch.branchName) ??
+        ({} as BranchCache);
+
+      if (
+        config.repositoryCache === 'enabled' &&
+        Object.keys(branchCache).length === 0
+      ) {
+        logger.debug(`No branch cache found for ${branch.branchName}`);
+      }
+    }
+    const branchManagersFingerprint = hasha(
+      branch.upgrades
+        .map((upgrade) => hashMap.get(upgrade.manager) ?? upgrade.manager)
+        .filter(is.string)
+    );
+    branch.branchFingerprint = hasha([
+      JSON.stringify(config),
+      branchManagersFingerprint,
+    ]);
```

Comment:
Move this inside the above if? (else)

Reply:
Will need to store the fngerpint inside cache whether branchExisted or not. Eg. We create a new branch then we need to add fingerpint to the cache to be later used.

Reply:
But what if the branch isn't created at all - I don't think it makes sense to store a fingerprint? Fingerprint is only valid if the branch exists, right?

Reply:
Ahh yes special edge case where the updates are generated but we don't create a branch. So this way we only store a fingerprint after the 2nd run.